showmethekey-gtk/smtk-keys-win.c

```diff
@@ -334,8 +335,16 @@ static void smtk_keys_win_init(SmtkKeysWin *win)
 	// Don't know why but realize does not work.
 	g_signal_connect(GTK_WIDGET(win), "map",
 			 G_CALLBACK(smtk_keys_win_on_map), NULL);
-}
 
+	GtkWindow* gtk_window = GTK_WINDOW(win);
+	// Create a layer shell surface instead of a normal floating window
+	gtk_layer_init_for_window(gtk_window);
+	// Make the layer appear on top of normal windows
+	gtk_layer_set_layer(gtk_window, GTK_LAYER_SHELL_LAYER_TOP);
+	gtk_layer_set_anchor(gtk_window, GTK_LAYER_SHELL_EDGE_BOTTOM, true);
+	gtk_layer_set_anchor(gtk_window, GTK_LAYER_SHELL_EDGE_RIGHT, true);
+	gtk_layer_set_anchor(gtk_window, GTK_LAYER_SHELL_EDGE_LEFT, true);
```

Comment:
Will this break desktops without layer shell? If the library don't check by itself, we may need to check here.

Reply:
I think it falls back to a floating window but i haven't checked

Reply:
it has a `gtk_layer_is_supported()` function though so it's easy to check